src/huggingface_hub/hf_api.py

```diff
@@ -3524,7 +3525,9 @@ def delete_repo(
             if not missing_ok:
                 raise
 
-    @validate_hf_hub_args
+    @_deprecate_method(
+        version="0.29.x", message="This method will be removed in v0.29.x. Please use `update_repo_settings` instead."
```

Comment:
```suggestion
        version="0.29", message="Use `update_repo_settings` instead."
```
nit: we can specify a shorter message for conciseness